### ISSUE TITLE
Mistake in detecting encoding of pom.xml

### ISSUE THREAD

**Issue by leveretconey** [openrewrite/rewrite-maven-plugin#734]
## What version of OpenRewrite are you using?
I am using

- OpenRewrite 8.13.1
- rewrite-maven-plugin 5.18.0
- Maven 3.5.0

## How are you running OpenRewrite?

pom.xml of the project: 

``` xml
<?xml version="1.0" encoding="UTF-8"?>
<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
  xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/maven-v4_0_0.xsd">
  <modelVersion>4.0.0</modelVersion>
  <groupId>org.example</groupId>
  <artifactId>debug</artifactId>
  <version>1.0-SNAPSHOT</version>
  <properties>
    <!-- UTF-8编码的中文-->
    <!-- ↑ Chinese characters in utf-8-->
    <project.build.sourceEncoding>gbk</project.build.sourceEncoding>
  </properties>
</project>

```

In a legacy application that java source files are using gbk encoding while pom.xml is using utf-8 encoding, OpenRewrite has recognized the file encoding of pom.xml as gbk. This leads to some problem in our scenerio.

After some debugging, I noticed that OpenRewrite reads `project.build.sourceEncoding` property from pom and sets it as the encoding of all files in this project. I think this behavior is not correct because this property is only applicable to source code files by Maven's design. 

## What did you expect to see?
pom.xml should be detected as using utf-8 encoding

## What did you see instead?
pom.xml is detected as using gbk encoding

## Are you interested in [contributing a fix to OpenRewrite](https://github.com/openrewrite/.github/blob/main/CONTRIBUTING.md#contributing-fixes)?
I won’t contribute to this because the code changes may be huge and involve multiple code repositories.

**Comment by leveretconey** [comment]
Thanks for your reply @timtebeek !

Let me add how we are currently using the MavenMojoProjectParser: Currently we set skipMavenParsing=true to skip the built-in POM parsing (otherwise sometimes it hangs, and we have never figured out the reason, which might be related to the Great Firewall), then we set parseAdditionalResources=true to treat the POM as a regular XML file. In addition, we have implemented some recipes to modify other configuration files that are outside of the src/ directory. Currently, it appears that these configuration files may also be affected by project.build.sourceEncoding, which could lead to incorrect encoding.

I have tested #735 and that does solve my issue.
